src/backend/gencomm.py

```diff
@@ -161,23 +161,25 @@ def populate_pupfns(pup_max_nesting, backend, blklens, builtin_types, builtin_ma
         yutils.display(OUTFILE, "\n")
 
     yutils.display(OUTFILE, "switch (type->kind) {\n")
-    for dtype1 in derived_types:
-        yutils.display(OUTFILE, "case YAKSI_TYPE_KIND__%s:\n" % dtype1.upper())
-        yutils.display(OUTFILE, "switch (type->u.%s.child->kind) {\n" % dtype1)
-        for dtype2 in derived_types:
-            yutils.display(OUTFILE, "case YAKSI_TYPE_KIND__%s:\n" % dtype2.upper())
-            yutils.display(OUTFILE, "rc = yaksuri_%si_populate_pupfns_%s_%s(type);\n" % (backend, dtype1, dtype2))
+    if pup_max_nesting > 0:
```

Comment:
Do we really want to support nesting = 0?

Reply:
Oh, I probably should manually test the code with ZE today. I haven't done it last night.

Reply:
> Nesting = 0 can be useful for dev since it only handles builtin types in backend and have frontend to handle all derived dtypes. So it builts really fast for dev purpose.
> 
> I would say we don't need to unsupport it.

OK, thanks. I didn't realize it still had some functionality.

Reply:
@raffenet Works fine with ZE, expected failures as the main branch.